src/components/forms/CloudDatabaseForm.js

```diff
@@ -0,0 +1,216 @@
+            key={this.state.use_data}
+            defaultToggled={this.state.use_data}
+            toggleFieldName="use_data"
+            onToggle={this.handleToggle}
+            className="fieldWidthSmallest"
+            id={`${this.props.data.name}-db-existing-instance`}
+          />
+        </IcseFormGroup>
+        <IcseFormGroup>
+          {/* Select Plan */}
+          <IcseSelect
+            labelText="Plan"
+            name="plan"
+            formName={this.props.data.name + "-db-plan"}
+            groups={this.state.service === "databases-for-mongodb" ? ["standard", "enterprise"] : ["standard"]}
+            value={this.state.plan}
+            handleInputChange={this.handleInputChange}
+            invalidText="Select a Plan."
+            className="fieldWidth"
+            id={`${this.props.data.name}-db-plan`}
+          />
+        {/* Select Service */}
+        <IcseSelect
+        labelText="Cloud Database"
+        name="service"
+        formName={this.props.data.name + "-db-service"}
+        groups={["databases-for-postgresql", "databases-for-etcd", "databases-for-redis", "databases-for-mongodb", "databases-for-mysql"]}
+        value={this.state.service}
+        handleInputChange={this.handleInputChange}
+        invalidText="Select a Cloud Database."
+        className="fieldWidth"
+        id={`${this.props.data.name}-db-service`}
+        />
+        {/* Select Group ID */}
+        <IcseSelect
+        labelText="Group ID"
+        name="group_id"
+        formName={this.props.data.name + "-db-groupId"}
+        groups={["member", "analytcs", "bi_connector", "search"]}
+        value={this.state.group_id}
+        handleInputChange={this.handleInputChange}
+        invalidText="Select a Group ID."
+        className="fieldWidth"
+        id={`${this.props.data.name}-db-groupId`}
+        />
+        </IcseFormGroup>
+        <IcseFormGroup>
+        {/* memory number input */}
+        <NumberInput
+            label="Memory"
+            id={this.props.data.name + "-db-memory"}
+            value={this.state.memory}
+            defaultValue={1}
+            max={112}
+            min={1}
+            onChange={this.handleInputChange}
+            name="memory"
+            hideSteppers={true}
+            invalidText="RAM must be a minimum of 1GB and a maximum 112GB per member"
+            className="fieldWidthSmaller leftTextAlign"
+          />
+        {/* disk number input */}
+        <NumberInput
+            label="Disk"
+            id={this.props.data.name + "-db-disk"}
+            value={this.state.disk}
+            defaultValue={1}
+            max={4096}
+            min={5}
+            onChange={this.handleInputChange}
+            name="disk"
+            hideSteppers={true}
+            invalidText="Disk must be a minimum of 5GB and a maximum 4096GB per member"
+            className="fieldWidthSmaller leftTextAlign"
+          />
+        {/* cpu text input */}
+        <IcseTextInput
```

Comment:
should cpus be NumberInput?

Reply:
CPU has a weird case where it can only be 3-28 dedicated cores, but if you'd like to use a shared CPU then you would put 0. I wasn't able to add proper logic for this case using NumberInput so I opted to using a TextInput and a callback. @evpirnia